### PR TITLE
add trpc v10

### PR DESCRIPTION
**Pull request by KATT**
## What does this PR do?

- Upgrades to tRPC 10
- Adds trpc with interop
- Add example procedure in backend + FE

**Notes:** 

- You can likely use and/or contribute to @sachinraja's codemod to speed up the migration, I did not try too much: https://github.com/sachinraja/trpc-v10-migrate-codemod/tree/main/src
- If 99% in your backend is "viewer" maybe you don't need the prefix & rather only prefix the public router? 

**Environment**: Staging(main branch) / Production

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [x] Test A
- [x] Test B

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't performed a self-review of my own code and corrected any misspellings
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/web/lib/app-providers.tsx [hunk 1]
```diff
@@ -54,7 +55,7 @@ const CustomI18nextProvider = (props: AppPropsWithChildren) => {
 };
 
 const AppProviders = (props: AppPropsWithChildren) => {
-  const session = trpc.useQuery(["viewer.public.session"]).data;
+  const session = proxy.public.session.useQuery().data;
```
Comment (KATT) [inline apps/web/lib/app-providers.tsx:58]: Example usage 

Notice that you can CMD+click `session` above and jump to the backend

**Review by sachinraja — APPROVED** [review]

**Comment by PeerRich** [comment]
great job already! anything we can help with to get this merged?

**Comment by KATT** [comment]
> great job already! anything we can help with to get this merged?

Ooo I just noticed that some e2e tests are not working, I'll try to fix those this week (or feel free to pick it up if you or @zomars feel inclinced)

File: apps/web/components/booking/pages/BookingPage.tsx [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { zodResolver } from "@hookform/resolvers/zod";
 import { EventTypeCustomInputType, WorkflowActions } from "@prisma/client";
 import { SchedulingType } from "@prisma/client";
+import { useMutation } from "@tanstack/react-query";
```
Comment (zomars) [inline apps/web/components/booking/pages/BookingPage.tsx:4]: Is there a reason this import is now coming from here? @KATT

File: packages/prisma/package.json [hunk 3]
```diff
@@ -10,9 +10,8 @@
     "db-nuke": "docker-compose down --volumes --remove-orphans",
     "db-reset": "run-s db-nuke db-setup",
     "db-seed": "yarn prisma db seed",
-    "db-setup": "run-s db-up db-deploy db-seed",
+    "db-setup": "run-s db-deploy db-seed",
     "db-studio": "yarn prisma studio",
-    "db-up": "docker-compose up -d",
```
Comment (zomars) [inline packages/prisma/package.json:15]: Why this was removed?

Comment (KATT) [reply packages/prisma/package.json:15]: Oops 0c8c15d is meant to be reverted

File: apps/web/components/booking/pages/BookingPage.tsx (hunk 2 above)
Comment (KATT) [reply apps/web/components/booking/pages/BookingPage.tsx:4]: https://tanstack.com/query/v4/docs/guides/migrating-to-react-query-4

File: packages/prisma/package.json (hunk 3 above)
Comment (zomars) [reply packages/prisma/package.json]: reverted now

**Review by zomars — APPROVED** [review]
Let's GOOO!

Lfg Lets Fucking Go

**Review by emrysal — APPROVED** [review]
LGTM 🚀 🚀 🚀

**Pull request merged**